internal/pkg/aws/ecs/ecs.go

```diff
@@ -149,6 +149,8 @@ func (e *ECS) waitUntilServiceStable(svc *Service) error {
 	for {
 		if len(svc.Deployments) == stableServiceDeploymentNum &&
 			aws.Int64Value(svc.DesiredCount) == aws.Int64Value(svc.RunningCount) {
+			// We can safely assume the service has been successfully updated, because before
+			// the circuit breaker is triggered to use the previous deployment, it should have timed out already.
```

Comment:
nit: to provide more context to other developers that may not understand why this comment may be needed 😛 
```suggestion
			// This conditional is sufficient to determine that the service is stable AND has successfully updated (hence not rolled-back).
			// The stable service cannot be a rolled-back service because a rollback can only be triggered by circuit breaker after ~1hr, 
			// by which time we would have already timed out.
```